src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,143 @@
+                String deadlineDate = deadlineDescriptionWithDate.split(" /by ")[1];
+                Deadline deadlineTask = new Deadline(deadlineDescription, deadlineDate);
+                listOfTasks.add(deadlineTask);
+                reply = ui.showDeadlineTask(deadlineTask, listOfTasks);
+                storage.writeToTextFile(listOfTasks);
+            } catch (StringIndexOutOfBoundsException e) {
+                return ":( OOPS!!! The description of a deadline cannot be empty.";
+            } catch (ArrayIndexOutOfBoundsException e) {
+                return ":( OOPS!!! You're missing some descriptions for your deadline.";
+            } catch (DateTimeParseException e) {
+                return ":( OOPS!!! You need to use yyyy-mm-dd for date format.";
+            }
+            break;
+
+        case EVENT:
+            try {
+                String eventDescriptionWithDate = command.substring(6);
+                String eventDescription = eventDescriptionWithDate.split(" /at ")[0];
+                String eventDate = eventDescriptionWithDate.split(" /at ")[1];
+                Event eventTask = new Event(eventDescription, eventDate);
+                listOfTasks.add(eventTask);
+                reply = ui.showEventTask(eventTask, listOfTasks);
+                storage.writeToTextFile(listOfTasks);
+            } catch (StringIndexOutOfBoundsException e) {
+                return ":( OOPS!!! The description of an event cannot be empty.";
+            } catch (ArrayIndexOutOfBoundsException e) {
+                return ":( OOPS!!! You're missing some descriptions for your event.";
+            } catch (DateTimeParseException e) {
+                return ":( OOPS!!! You need to use yyyy-mm-dd for date format.";
+            }
+            break;
+
+        case DELETE:
+            try {
+                int deleteIndex = Integer.parseInt(response[1]) - 1;
+                Task deletedTask = listOfTasks.getTask(deleteIndex);
+                listOfTasks.remove(deleteIndex + 1);
+                reply = ui.showDeletedTask(deletedTask, listOfTasks);
+                storage.writeToTextFile(listOfTasks);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                return ":( OOPS!!! You're missing an index for delete.";
+            }
+            break;
+
+        case FIND:
+            try {
+                String keyword = response[1];
+                TaskList matchingTasks = new TaskList();
+                for (Task task : listOfTasks.getListOfTasks()) {
+                    if (task.getDescription().contains(keyword)) {
+                        matchingTasks.add(task);
+                    }
+                }
+                reply = ui.showFindTask(matchingTasks);
+            } catch (ArrayIndexOutOfBoundsException e) {
+                return ":( OOPS!!! The description of a find cannot be empty.";
+            }
+            break;
+
+        case OTHERS:

```

Comment:
If I recall correctly, it is recommended to just use default for the final enumeration case, not that it would really make a difference but perhaps for consistent coding style? :")